src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c

```diff
@@ -22,141 +57,1577 @@ const ActorInit En_Goroiwa_InitVars = {
+                             ((Rand_ZeroOne() * 150.0f) + 250.0f) * arg5, phi_s3, 0, 0x36, -1, 0xEF, phi_s1);
+
+        spE8.x += ((Rand_ZeroOne() * 800.0f) - 400.0f) * arg5;
+        spE8.y += ((Rand_ZeroOne() * 800.0f) - 250.0f) * arg5;
+        spE8.z += ((Rand_ZeroOne() * 800.0f) - 400.0f) * arg5;
+
+        func_800B0E48(globalCtx, &spE8, &D_80942E48, &D_80942E54, arg3, arg4, (Rand_ZeroOne() * 60.0f) + spAC,
+                      (Rand_ZeroOne() * 30.0f) + spAC);
+    }
+}
+
+void func_80940E38(EnGoroiwa* this, GlobalContext* globalCtx) {
+    f32 sp5C;
+    s32 pad;
+    f32 sp54;
+    Vec3f sp48;
+    s16 sp46;
+    s16 temp_a0;
+
+    if (this->actor.flags & 0x40) {
+        if (this->actor.xzDistToPlayer < 1000.0f) {
+            sp5C = (1000.0f - this->actor.xzDistToPlayer) * 0.0012f * (this->actor.speedXZ * 0.1f);
+            if (Rand_ZeroOne() < sp5C) {
+                this->unk_1CE += 20000;
+                sp46 = (s32)Rand_ZeroFloat(20000.0f) + this->unk_1CE;
+                temp_a0 = sp46 - this->actor.world.rot.y;
+                if (ABS(temp_a0) < 0x4000) {
+                    sp54 = Math_CosS(temp_a0) * 1.6f * this->unk_1DC;
+                } else {
+                    sp54 = this->unk_1DC;
+                }
+
+                sp48.x = (Math_SinS(sp46) * sp54) + this->actor.world.pos.x;
+                sp48.y = this->actor.world.pos.y + 20.0f;
+                sp48.z = (Math_CosS(sp46) * sp54) + this->actor.world.pos.z;
+
+                func_800B0E48(globalCtx, &sp48, &D_80942E48, &D_80942E54, &D_80942E38, &D_80942E44,
+                              (Rand_ZeroOne() * 600.0f) + (600.0f * (this->actor.scale.x + 0.1f) * 0.5f),
+                              (s32)(Rand_ZeroOne() * 50.0f) + 30);
+            }
+        }
+    }
+}
+
+void func_80941060(EnGoroiwa* this, GlobalContext* globalCtx) {
+    Vec3f spAC;
+    Vec3f spA0;
+    Vec3f sp94;
+    Vec3s* vec = &this->collider.elements[0].info.bumper.hitPos;
+    s32 i;
+
+    for (i = 0; i < 4; i++) {
+        sp94.x = ((Rand_ZeroOne() * 14.0f) - 7.0f) + vec->x;
+        sp94.y = ((Rand_ZeroOne() * 14.0f) - 7.0f) + vec->y;
+        sp94.z = ((Rand_ZeroOne() * 14.0f) - 7.0f) + vec->z;
+
+        spA0.x = (Rand_ZeroOne() - 0.5f) * 1.6f;
+        spA0.y = -0.8f;
+        spA0.z = (Rand_ZeroOne() - 0.5f) * 1.6f;
+
+        spAC.x = spA0.x * -0.06f;
+        spAC.y = spA0.y * -0.06f;
+        spAC.z = spA0.z * -0.06f;
+
+        func_800B0E48(globalCtx, &sp94, &spA0, &spAC, &D_80942E38, &D_80942E44, (s32)(Rand_ZeroOne() * 30.0f) + 15,
+                      (s32)(Rand_ZeroOne() * 40.0f) + 30);
+    }
+}
+
+void func_80941274(EnGoroiwa* this, GlobalContext* globalCtx) {
+    Audio_PlaySoundAtPosition(globalCtx, &this->actor.world.pos, 30, NA_SE_EV_SNOWBALL_BROKEN);
+}
+
+void EnGoroiwa_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnGoroiwa* this = THIS;
+    f32 temp_f0;
+    s32 params = ENGOROIWA_GET_FF(&this->actor);
+    s32 sp28 = params * 8;
```

Comment:
I think since params are generally used as bit flags and not actual numbers, I would prefer this to be shifted left 3, instead of multiplied by 8. However don't feel too strongly about it.

Reply:
Does not match.